lib/Share/Helper/ShareAPIController.php

```diff
@@ -78,14 +78,14 @@ public function formatShare(IShare $share): array {
 	 * @param IShare $share
 	 * @param string $shareWith
 	 * @param int $permissions
-	 * @param string $expireDate
+	 * @param ?string $expireDate
 	 * @throws OCSNotFoundException
 	 */
-	public function createShare(IShare $share, string $shareWith, int $permissions, string $expireDate): void {
+	public function createShare(IShare $share, string $shareWith, int $permissions, ?string $expireDate = null): void {
 		$share->setSharedWith($shareWith);
 		$share->setPermissions($permissions);
 
-		if ($expireDate !== '') {
+		if ($expireDate !== null && $expireDate !== '') {
```

Comment:
`empty` returns true for both `null` and empty string (`''`).

```suggestion
		if (!empty($expireDate)) {
```

Reply:
Yes, but I wanted to keep it separated to explicitly show 2 different cases, because `''` and `null` have different semantics here